### DISCUSSION TITLE
Where are the "Storage Assets Presets"?

### DISCUSSION THREAD

**Discussion by yii-joblo** [directus/directus#3701] [Archive]
I can't find my file presets on the local disk. Where is my "square"-preset.
![image](https://user-images.githubusercontent.com/1770823/104696076-fade5c00-570d-11eb-95fc-830a4cc057c0.png)

In the api from version 9 I missing the presets/thumnails and the url:
![Bildschirmfoto 2021-01-15 um 08 48 55](https://user-images.githubusercontent.com/1770823/104697728-7b9e5780-5710-11eb-8c0a-e909db2225ee.png)

Directus 8 - it's more clear:
![Bildschirmfoto 2021-01-15 um 08 51 05](https://user-images.githubusercontent.com/1770823/104697678-69bcb480-5710-11eb-921a-4061b9e79464.png)

**Comment by yii-joblo** [comment]
I need some like this:
`
const getFileUrl = (id, preset = null) => {
    return API_SERVER + '/assets/' + id <---------------- + preset?? 
}`

**Reply by rijkvanzanten** [reply]
Here you go!

```js
const getFileUrl = (id, preset) => `${API_SERVER}/assets/${id}?key=${preset}`
```

**Reply by yii-joblo** [reply]
ok, it's the same as in v8 :-)
But the key is not included in filename for folder uploads.
And what is the key of the directus default presets? Where can I get the existing keys?

![image](https://user-images.githubusercontent.com/1770823/104746506-abb91b00-574f-11eb-958c-ec1863ee2def.png)

directus-small-crop, as in v8, is not a valid key:

![image](https://user-images.githubusercontent.com/1770823/104746983-40237d80-5750-11eb-872a-ae247da69cb1.png)

**Reply by rijkvanzanten** [reply]
The files are stored based on their properties (eg w/h/fit/etc) instead of key to avoid duplicates. 

The system keys can be found here:

https://github.com/directus/directus/blob/596230107a4a57b40460aec59cc08a5516efde0b/api/src/constants.ts#L3-L37

**Reply by yii-joblo** [reply]
thx

**Comment by benhaynes** [comment]
@rijkvanzanten — any reason we can't get this info (full asset paths and array of all available thumbnails) in the v9 API?

**Reply by rijkvanzanten** [reply]
We discuss that in a new discussion 👍🏻
